### PR TITLE
MINOR: Remove APIs deprecated in 0.11.0 for core and clients

### PR DESCRIPTION
**Pull request by ijuma**
Not included: old consumers and checksum methods

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by omkreddy** [comment]
LGTM

**Comment by ijuma** [comment]
@lindong28 does this look OK to you?

File: clients/src/main/java/org/apache/kafka/clients/producer/RecordMetadata.java [hunk 1]
```diff
@@ -56,16 +56,6 @@ public RecordMetadata(TopicPartition topicPartition, long baseOffset, long relat
...
 
-    /**
-     * @deprecated As of 0.11.0. Use @{@link RecordMetadata#RecordMetadata(TopicPartition, long, long, long, Long, int, int)}.
-     */
-    @Deprecated
-    public RecordMetadata(TopicPartition topicPartition, long baseOffset, long relativeOffset, long timestamp,
-                          long checksum, int serializedKeySize, int serializedValueSize) {
-        this(topicPartition, baseOffset, relativeOffset, timestamp, Long.valueOf(checksum), serializedKeySize,
-                serializedValueSize);
-    }
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/clients/producer/RecordMetadata.java:67]: @hachikuji any concerns about removing this?

**Comment by lindong28** [comment]
@ijuma Thanks! LGTM.

**Comment by ijuma** [comment]
Merged to trunk and 2.0.

**Pull request merged**